src/library/trackset/setlogfeature.cpp

```diff
@@ -360,17 +410,99 @@ void SetlogFeature::slotJoinWithPrevious() {
                          << " previous:" << previousPlaylistId;
                 if (m_playlistDao.copyPlaylistTracks(
                             currentPlaylistId, previousPlaylistId)) {
-                    m_lastRightClickedIndex = constructChildModel(previousPlaylistId);
                     m_playlistDao.deletePlaylist(currentPlaylistId);
-                    reloadChildModel(previousPlaylistId); // For moving selection
-                    emit showTrackModel(m_pPlaylistTableModel);
-                    activatePlaylist(previousPlaylistId);
                 }
             }
         }
     }
 }
 
+void SetlogFeature::slotLockAllChildPlaylists() {
+    lockOrUnlockAllChildPlaylists(true);
+}
+
+void SetlogFeature::slotUnlockAllChildPlaylists() {
+    lockOrUnlockAllChildPlaylists(false);
+}
+
+void SetlogFeature::lockOrUnlockAllChildPlaylists(bool lock) {
+    if (!m_lastRightClickedIndex.isValid()) {
+        return;
+    }
+    if (lock) {
+        qWarning() << "lock all child playlists of" << m_lastRightClickedIndex.data().toString();
+    } else {
+        qWarning() << "unlock all child playlists of" << m_lastRightClickedIndex.data().toString();
+    }
+    // Ask for confirmation: "Delete all playlists in this group, including locked?"
+    TreeItem* item = static_cast<TreeItem*>(m_lastRightClickedIndex.internalPointer());
+    if (!item) {
+        return;
+    }
+    const QList<TreeItem*> yearChildren = item->children();
+    if (yearChildren.isEmpty()) {
+        return;
+    }
+
+    QSet<int> ids;
+    for (auto* child : qAsConst(yearChildren)) {
+        bool ok = false;
+        int childId = child->getData().toInt(&ok);
+        if (ok && childId != kInvalidPlaylistId) {
+            ids.insert(childId);
+        }
+    }
+    m_playlistDao.setPlaylistsLocked(ids, lock);
+}
+
+void SetlogFeature::slotDeleteAllChildPlaylists() {
+    if (!m_lastRightClickedIndex.isValid()) {
+        return;
+    }
+    qWarning() << "slotDeleteAllChildPlaylists of" << m_lastRightClickedIndex.data().toString();
+    // Ask for confirmation: "Delete all playlists in this group, including locked?"
+    TreeItem* item = static_cast<TreeItem*>(m_lastRightClickedIndex.internalPointer());
+    if (!item) {
+        return;
+    }
+    const QList<TreeItem*> yearChildren = item->children();
+    if (yearChildren.isEmpty()) {
+        return;
+    }
+
+    QMessageBox::StandardButton btn = QMessageBox::question(nullptr,
+            tr("Confirm Deletion"),
+            tr("Do you really want to delete all playlist in <b>%1</b>?<br><br>"
+               "Note: this also includes locked playlists!")
+                    .arg(m_lastRightClickedIndex.data().toString()),
+            QMessageBox::Yes | QMessageBox::No,
+            QMessageBox::No);
+    if (btn == QMessageBox::No) {
+        return;
+    }
+    // Double-check, this is a weighty decision
+    btn = QMessageBox::warning(nullptr,
```

Comment:
This is a severe action, though, in my experience you can get used to confirmation boxes and automatically hit Yes.
There's rarely/never a second confirmation required, so it's almost guarantueed this one gets the attention it deserves.

Do you have another solution in mind? (undo is not an option, yet)

Reply:
I think if we have the default action "No" and the locks are respected we can go with a single confirmation dialog. 

But I am actually not entirely against these two dialogs. 

My idea was to use something cool instead. 
Like three buttons and a request "To confirm press the button in the following order ... "
Or an edit box for "yes" or a captcha. 

But of cause it is extra work we night want to avoid.